### PR TITLE
FF122 Date parser parses first three chars of month

### PR DESCRIPTION
**Pull request by hamishwillee**
For parity with Chrome/Safari the data parsers now only looks at the first three characters of the month when parsing non-standard date strings. Previously it would parse only valid truncated values of the month name - e.g. for december it might accept Dec, Dece, Decemb, December. Now it will accept DecFooo like the other browsers. 

This adds a release note and an example in the `Date.parse()` function.

Related docs work can be tracked in #31930

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md [hunk 1]
```diff
@@ -94,6 +94,14 @@ Date.parse("2014-02-30");
 // NaN in Safari and Firefox;
 // 1393718400000 in Chrome (Sun Mar 02 2014 00:00:00 GMT+0000)
 Date.parse("02/30/2014"); // 1393718400000 in all implementations
+
+// Chrome, Safari, and Firefox 122 and later parse only the first three letters for the month.
+// FF121 and earlier parse first three letters and any substring up to the correct month name.
+Date.parse("04 Dec 1995"); // Valid all implementations
```
Comment (bsmth) [inline files/en-us/web/javascript/reference/global_objects/date/parse/index.md:100]: The others above show the UTC timestamp, not sure if you want to do the same

```suggestion
Date.parse("04 Dec 1995"); // 818031600000 in all implementations
```

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md [hunk 2]
```diff
@@ -94,6 +94,14 @@ Date.parse("2014-02-30");
 // NaN in Safari and Firefox;
 // 1393718400000 in Chrome (Sun Mar 02 2014 00:00:00 GMT+0000)
 Date.parse("02/30/2014"); // 1393718400000 in all implementations
+
```
Comment (bsmth) [inline files/en-us/web/javascript/reference/global_objects/date/parse/index.md:97]: Out of scope of this PR, but I think this section is getting a little disorganized. Maybe in future we can separate this code block into categories or subsections.

**Review by bsmth — APPROVED** [review]
Thanks Hamish! Some comments for consideration, but leaving a +1 👍🏻

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md [hunk 3]
```diff
@@ -94,6 +94,14 @@ Date.parse("2014-02-30");
...
+// Chrome, Safari, and Firefox 122 and later parse only the first three letters for the month.
+// FF121 and earlier parse first three letters and any substring up to the correct month name.
+Date.parse("04 Dec 1995"); // Valid all implementations
+Date.parse("04 Decem 1995"); // Valid all implementations
+Date.parse("04 December 1995"); // Valid all implementations
+Date.parse("04 DecFoo 1995"); // NaN in Firefox 121 and earlier. Valid otherwise.
+Date.parse("04 De 1995"); // NaN in all implementations
```
Comment (hamishwillee) [inline files/en-us/web/javascript/reference/global_objects/date/parse/index.md:104]: Matching https://github.com/mdn/content/pull/31931/files#r1467403276

```suggestion
Date.parse("04 Decem 1995"); // 818031600000 in all implementations
Date.parse("04 December 1995"); // 818031600000 in all implementations
Date.parse("04 DecFoo 1995"); // NaN in Firefox 121 and earlier. 818031600000 in other implementations
Date.parse("04 De 1995"); // NaN in all implementations
```

File: files/en-us/web/javascript/reference/global_objects/date/parse/index.md (hunk 1 above)
Comment (hamishwillee) [reply files/en-us/web/javascript/reference/global_objects/date/parse/index.md:100]: I do :-)

**Comment by hamishwillee** [comment]
Thanks for the review @bsmth - much appreciated.

**Pull request merged**